cordova-plugin-one/src/android/OnePlugin.java

```diff
@@ -8,419 +8,451 @@
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    try {
-                        String token = one.getPushToken();
-                        callbackContext.success(token);
-                    } catch (Exception e) {
-                        callbackContext.error("Failed to get push token");
-                    }
-                }
-            });
-            return true;
-        } else if ("enablePushNotifications".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    boolean enable = args.optBoolean(0);
-                    one.enablePushNotifications(enable);
-                    callbackContext.success();
-                }
-            });
-            return true;  
-        } else if ("whitelistIdentityTransferLinks".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    ArrayList<String> links = getStringListFromCordovaArgs(args);
-                    if (links.size() > 0) {
-                        one.whitelistIdentityTransferLinks(links);
-                        callbackContext.success();
-                    } else {
-                        callbackContext.error("The SDK cannot whitelist any links as no links were provided.");
-                    }
-                }
-            });
-            return true;
-        } else if ("blacklistIdentityTransferLinks".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    ArrayList<String> links = getStringListFromCordovaArgs(args);
-                    if (links.size() > 0) {
-                        one.blacklistIdentityTransferLinks(links);
-                        callbackContext.success();
-                    } else {
-                        callbackContext.error("The SDK cannot blacklist any links as no links were provided.");
-                    }
-                }
-            });
-            return true;
+  protected static final String LOG_TAG = "OnePlugin";
+
+  @Override
+  public boolean execute(final String action, final CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
+    if ("initializeOne".equals(action)) {
+      return initializeOne(args, callbackContext);
+    } else if ("sendInteraction".equals(action)) {
+      cordova.getActivity().runOnUiThread(() -> {
+        try {
+          JSONObject jsonObject = args.optJSONObject(1);
+          HashMap<String, String> propertiesMap = getPropertiesFromJSONObject(jsonObject);
+          sendInteraction(args.getString(0), propertiesMap, callbackContext);
+        } catch (Exception e) {
+          e.printStackTrace();
+          callbackContext.error("Failed to send properties: " + e.getLocalizedMessage());
```

Comment:
Do we need to tag the logs or is it pretty obvious they are coming from our plugin?

Reply:
Good call, I'll add a tag and use Log.e instead.